crates/ruff_server/src/message.rs

```diff
@@ -0,0 +1,63 @@
+use std::sync::OnceLock;
+
+use lsp_types::notification::Notification;
+
+use crate::server::ClientSender;
+
+static MESSENGER: OnceLock<ClientSender> = OnceLock::new();
```

Comment:
The once lock is clever. The only downside is that once locks can be tricky when writing tests. Is it difficult to get access to the `ClientSender`? If not, what do you think of mirroring the `write!` macro where the first argument is the output stream:

```rust
show_message_err!(sender, "My message {var}")
```

Reply:
I don't think it's worthwhile to make the API more involved just for the purposes of testing. If we do add tests in the future, we can rework `show_message` to take a reference to a client sender and then test `show_message` directly with a mock sender.